# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/svg/SVGAnimatedLength.h
# Commit: 5158484e9eab
# Full Hash: 5158484e9eabc55a48e73e68ad4b6e676d057997
# Author: Robert Longson <longsonr@gmail.com>
# Date: 2023-09-12 04:12:49
# Regressor Bug: 1287054
# File Overlap Count: 1
# Description:
#   Bug 1287054 part 2 - support vw, vh, vmin and vmax units for non-CSS lengths r=emilio,jgilbert
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D180619
# ==============================================================================

diff -r 35034282dc7c -r 5158484e9eab dom/svg/SVGAnimatedLength.h
--- a/dom/svg/SVGAnimatedLength.h	Mon Sep 11 18:20:18 2023 +0000
+++ b/dom/svg/SVGAnimatedLength.h	Mon Sep 11 18:20:19 2023 +0000
@@ -42,6 +42,7 @@
   static GeckoFontMetrics DefaultFontMetrics();
   static GeckoFontMetrics GetFontMetrics(const Element* aElement);
   static WritingMode GetWritingMode(const Element* aElement);
+  static CSSSize GetCSSViewportSizeFromContext(const nsPresContext* aContext);
 
   virtual ~UserSpaceMetrics() = default;
 
@@ -51,6 +52,7 @@
   float GetIcWidth(Type aType) const;
   float GetCapHeight(Type aType) const;
   virtual float GetAxisLength(uint8_t aCtxType) const = 0;
+  virtual CSSSize GetCSSViewportSize() const = 0;
 
  protected:
   virtual GeckoFontMetrics GetFontMetricsForType(Type aType) const = 0;
@@ -72,6 +74,7 @@
     return SVGContentUtils::GetFontSize(GetElementForType(aType));
   }
   float GetAxisLength(uint8_t aCtxType) const override;
+  CSSSize GetCSSViewportSize() const override;
 
  private:
   bool EnsureCtx() const;
@@ -89,6 +92,7 @@
 
   float GetEmLength(Type aType) const override;
   gfx::Size GetSize() const override;
+  CSSSize GetCSSViewportSize() const override;
 
  private:
   GeckoFontMetrics GetFontMetricsForType(Type aType) const override;